internal/w32/shell_item_windows.go

```diff
@@ -56,7 +59,7 @@ func (obj *ShellItem) DisplayName() string {
 		return ""
 	}
 	defer CoTaskMemFree(uintptr(unsafe.Pointer(p)))
-	return syscall.UTF16ToString((*[1 << 30]uint16)(unsafe.Pointer(p))[:])
+	return syscall.UTF16ToString(unsafe.Slice(p, maxUint16Array))
```

Comment:
This looks like it is missing the cast to `(*uint16)`.

Reply:
Nope, the `var p *uint16` takes care of that already.